tests/integration/api/test_api.py

```diff
@@ -1,478 +1,301 @@
 # python
 import datetime
+import json
 import os
 import time
 import unittest
-import requests
-import json
 
-# 3p
-import pytest
+import requests
 
 # datadog
-from datadog import initialize
 from datadog import api as dog
-from datadog.api.exceptions import ApiError
-from tests.integration.util.snapshot_test_utils import (
-    assert_snap_not_blank, assert_snap_has_no_events
-)
-
-TEST_USER = os.environ.get('DD_TEST_CLIENT_USER')
-API_KEY = os.environ.get('DD_TEST_CLIENT_API_KEY', "a"*32)
-APP_KEY = os.environ.get('DD_TEST_CLIENT_APP_KEY', "a"*40)
-API_HOST = os.environ.get('DATADOG_HOST')
-FAKE_PROXY = {
-    "https": "http://user:pass@10.10.1.10:3128/",
-}
-
-
-def eq(a, b):
-    assert a == b
-
-
-def ok(assertion, message):
-    assert assertion, message
+from datadog import initialize
+from tests.integration.util.snapshot_test_utils import assert_snap_has_no_events, assert_snap_not_blank
+
+TEST_USER = os.environ.get("DD_TEST_CLIENT_USER")
+API_KEY = os.environ.get("DD_TEST_CLIENT_API_KEY", "a" * 32)
+APP_KEY = os.environ.get("DD_TEST_CLIENT_APP_KEY", "a" * 40)
+API_HOST = os.environ.get("DATADOG_HOST")
+FAKE_PROXY = {"https": "http://user:pass@10.10.1.10:3128/"}
+WAIT_TIME = 10
+
+
+def get_with_retry(
+    resource_type,
+    resource_id=None,
+    operation="get",
+    retry_count=10,
+    retry_condition=lambda r: r.get("errors"),
+    **kwargs
+):
+    if resource_id is None:
+        resource = getattr(getattr(dog, resource_type), operation)(**kwargs)
+    else:
+        resource = getattr(getattr(dog, resource_type), operation)(resource_id, **kwargs)
+    retry_counter = 0
+    while retry_condition(resource) and retry_counter < 10:
+        if resource_id is None:
+            resource = getattr(getattr(dog, resource_type), operation)(**kwargs)
+        else:
+            resource = getattr(getattr(dog, resource_type), operation)(resource_id, **kwargs)
+        retry_counter += 1
+        time.sleep(WAIT_TIME)
```

Comment:
should we raise an exception here if our retry condition is still what we got back? Currently it'll fail at the test itself when an assertion happens on this data but seems like we could fail faster here.

Reply:
good point